### PR TITLE
refactor: field by field value

### PR DESCRIPTION
**Pull request by jeqo**
Apply some refactoring and clarify previous suggestions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -141,19 +141,23 @@ transforms=MakeTombstone
 transforms.MakeTombstone.type=io.aiven.kafka.connect.transforms.MakeTombstone
 ```
 
-### `FilterByValueRegex`
+### `FilterByFieldValue`
```
Comment (jeqo) [inline README.md:144]: Took a step back about focusing only on regex. A simpler comparison could save us some cycles at runtime.

File: src/main/java/io/aiven/kafka/connect/transforms/FilterByFieldValue.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    private Predicate<Optional<String>> filterCondition;
+
+    @Override
+    public R apply(final R record) {
+        if (record.value() instanceof Struct) {
+            return handleStruct(record);
+        } else if (record.value() instanceof Map) {
+            return handleMap(record);
+        }
+        return record; // if record is other than map or struct, pass-by
```
[the remark this answers is not in the document]
Comment (sebinsunny) [reply src/main/java/io/aiven/kafka/connect/transforms/FilterByFieldValue.java:48]: now it is clear

File: src/main/java/io/aiven/kafka/connect/transforms/FilterByFieldValue.java [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        if (fieldValue instanceof String
+                || fieldValue instanceof Long
+                || fieldValue instanceof Integer
+                || fieldValue instanceof Short
+                || fieldValue instanceof Double
+                || fieldValue instanceof Float
+                || fieldValue instanceof Boolean) {
+            text = Optional.of(fieldValue.toString());
+        }
+        return text;
```
[the remark this answers is not in the document]
Comment (sebinsunny) [reply src/main/java/io/aiven/kafka/connect/transforms/FilterByFieldValue.java:93]: Looks good, we don't need to add Byte

File: src/test/java/io/aiven/kafka/connect/transforms/FilterByFieldValueTest.java [hunk 4]
```diff
@@ -127,68 +124,22 @@ void shouldFilterOutRecordsNotEqualsReadEvents() {
         assertNull(actual, "Record with op not equal to 'r' should be filtered out");
     }
 
-    @Test
-    void shouldFilterMatchingArrayFieldValue() {
```
[the remark this answers is not in the document]
Comment (sebinsunny) [reply src/test/java/io/aiven/kafka/connect/transforms/FilterByFieldValueTest.java:131]: ok, got it

File: README.md (hunk 1 above)
Comment (sebinsunny) [reply README.md:144]: ok

**Pull request merged**